### ISSUE TITLE
Plugin configuration is not visible when WordPress is configured to use stateless mode 

### ISSUE THREAD

**Issue by wysow** [OneSignal/OneSignal-WordPress-Plugin#33]
I discovered that the plugin configuration pages are not visible when disabling modification on files and plugins and core automatic updates in WordPress. 

This is needed when you host your blog on a stateless platform like AWS Elastic Beanstalk.

**Comment by jasonpang** [comment]
Hi @wysow,

Do you know if other plugins are experiencing a similar issue or if its just ours?

**Comment by wysow** [comment]
@jasonpang AFAIK it's the only plugin doing this in the plugin used on the website I'm working on... Do you store configuration settings in a file instead of database? That's the only thing I can think of right now...

**Comment by jasonpang** [comment]
Hey @wysow,

We only store configuration files in the database. I'm not sure off the top of my head why this would be happening either. Will have to look into it soon. How are you guys able to use the plugin at the moment with this issue?

**Comment by wysow** [comment]
@jasonpang As the database is in AWS RDS and the plugin was installed and configured before we migrate the website on AWS Elastic Beanstalk the plugin configuration was already done and working... ;)

**Comment by wysow** [comment]
@jasonpang any news on this one? Just to add more information when I say "stateless mode" I mean that I added `define('DISALLOW_FILE_MODS', true);` and `define('WP_AUTO_UPDATE_CORE', false);` in `wp-config.php`. I did update the plugin to the last release and same behavior.

**Comment by wysow** [comment]
@jasonpang another information I just discovered is that on page `/wp-admin/admin.php?page=onesignal-push` I got error message that I do not have the rights to see this page, but my account is a full admin of the wordpress...

**Comment by wysow** [comment]
Here seems to be the problem https://github.com/OneSignal/OneSignal-WordPress-Plugin/blob/master/onesignal-admin.php#L74

**Comment by wysow** [comment]
@jasonpang If you can tell me why you did this check, I can try to make a PR to optimize this...

**Comment by wysow** [comment]
@jasonpang Some news on this one?

**Comment by jasonpang** [comment]
Hi @wysow,

Sorry for the delay in getting back to you. I'm not exactly sure of the reasons we check for whether the user has the capability to update plugins. We can maybe remove this as I don't think we've run into any issues with permissions before. Did you have any other ideas on optimizing it?

**Comment by wysow** [comment]
@jasonpang Nope it sounds like it will resolve the main problem without creating new ones... ;)

**Comment by jasonpang** [comment]
Hi @wysow,

Sorry this took so long. We now use a different check to detect whether the user can view the OneSignal Push page and save configuration settings. As before, the user must be an administrator, but the capability check for whether the user is an administrator is `current_user_can('delete_users')` instead of `current_user_can('update_plugins')`, which does fail when adding:

```
define('DISALLOW_FILE_MODS', true);
define('WP_AUTO_UPDATE_CORE', false);
```

to `wp-config.php`.

You guys have probably impemented this workaround for yourselves by now, but I just wanted to let you know that we did get around to doing it eventually! Cheers, and sorry for the delay.

Fixed by: https://github.com/OneSignal/OneSignal-WordPress-Plugin/commit/884da1cb7ef557484ee9c7e0f9e9d9d8fa87c0a4
